datafusion/core/src/datasource/listing/helpers.rs

```diff
@@ -241,6 +241,7 @@ async fn prune_partitions(
     partitions: Vec<Partition>,
     filters: &[Expr],
     partition_cols: &[(String, DataType)],
+    ctx: &SessionState,
```

Comment:
I recommend passing `&ConfigOptions` directly here as the function only needs the config, not the entire session state